src/goto-symex/symex_dead.cpp

```diff
@@ -45,14 +46,34 @@ void goto_symext::symex_dead(statet &state)
     state.value_set.assign(ssa, rhs, ns, true, false);
   }
 
-  ssa_exprt ssa_lhs=to_ssa_expr(ssa);
-  const irep_idt &l1_identifier=ssa_lhs.get_identifier();
-
-  // prevent propagation
-  state.propagation.erase(l1_identifier);
+  const exprt l2_fields = field_sensitivityt::get_fields(ns, ssa);
+  std::set<exprt> l2_fields_set;
+  find_symbols(l2_fields, l2_fields_set);
 
-  // L2 renaming
-  auto level2_it = state.level2.current_names.find(l1_identifier);
-  if(level2_it != state.level2.current_names.end())
-    symex_renaming_levelt::increase_counter(level2_it);
+  for(const exprt &l2_field_set : l2_fields_set)
+  {
+    ssa_exprt ssa_lhs = to_ssa_expr(l2_field_set);
+    const irep_idt &l1_identifier = ssa_lhs.get_identifier();
+
+    // prevent propagation
+    state.propagation.erase(l1_identifier);
+
+    // TODO: not sure why the rest below is necessary
```

Comment:
Then let's remove it. If no test fails then it is not needed, no? ;)

Reply:
#3980 actually clarifies those comments. (This PR just changes the indent, no actual code changes here.)